source/connection-monitoring-and-pooling/tests/pool-checkout-custom-maxConnecting-is-enforced.yml

```diff
@@ -0,0 +1,54 @@
+version: 1
+style: integration
+description: custom maxConnecting is enforced
+runOn:
+  -
+    minServerVersion: "4.4.0"
+failPoint:
+  configureFailPoint: failCommand
+  mode: "alwaysOn"
+  data:
+    failCommands: ["isMaster","hello"]
+    closeConnection: false
+    blockConnection: true
+    blockTimeMS: 500
+poolOptions:
+  maxConnecting: 1
+  # gives opportunity for the checkout in thread2 to establish a new connection, which it must not do until thread1 establishes one
+  maxPoolSize: 2
+  waitQueueTimeoutMS: 5000
+operations:
+  - name: ready
+  # thread1 exists to consume the single permit to open a connection,
+  # so that thread2 would be blocked acquiring a permit, which results in ordering its ConnectionCreated event after
+  # the ConnectionReady event from thread1.
+  - name: start
+    target: thread1
+  - name: start
+    target: thread2
+  - name: checkOut
+    thread: thread1
+  - name: waitForEvent
+    event: ConnectionCreated
+    count: 1
+  # Wait long enough to be sure that thread1 started establishing a connection,
+  # and short enough to be sure that it is still blocked by failPoint.
+  - name: wait
```

Comment:
I think we can omit this wait, since observing the `ConnectionCreatedEvent` should be enough to ensure that the next thread cannot create a new connection per the psuedocode in [Creating a Connection (internal implementation)](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#creating-a-connection-internal-implementation).

Reply:
Indeed, we of course grab a permit before emitting `ConnectionCreatedEvent`. Thanks for spotting this 👍

Reply:
Done.